### ISSUE TITLE
[BUG] Asset name is rendered as empty string in Data Docs

### ISSUE THREAD

**Issue by jschra** [great-expectations/great_expectations#10339]
**Describe the bug**
When using checkpoint to validate a pandas data asset based on an in-memory data frame, the resulting validation result rendered in the Data Docs website has an empty asset name.

**To Reproduce**
To reproduce the error, you can run this single script in its entirety along with the data file I have attached to this post. Within it, a new file based context will be created with all default values, so there is no need to share the great_expectations.yml file.

```python
import great_expectations as gx
import pandas as pd

# -- Set GX constants for artifact creation
NAME_DATA_SOURCE = "pandas"
NAME_DATA_ASSET = "tutorial_data"
NAME_BATCH_DEF = "pandas_tutorial"
NAME_EXPECTATION_SUITE = "tutorial"
NAME_VALIDATION_DEF = "pandas-validation"
NAME_CHECKPOINT = "pandas"

# -- Load data for configuration
FILE_CONFIGURE = "data/yellow_tripdata_2021-11.csv"
df_configure = pd.read_csv(FILE_CONFIGURE)

# -- 1. Initialize GX for configuration
context = gx.get_context(mode="file")

data_source = context.data_sources.add_pandas(name=NAME_DATA_SOURCE)
data_asset = data_source.add_dataframe_asset(name=NAME_DATA_ASSET)
batch_definition = data_asset.add_batch_definition_whole_dataframe(NAME_BATCH_DEF)

# -- 2. Configure expectation suite to be called over runtime data later
expectation_suite = gx.ExpectationSuite(name=NAME_EXPECTATION_SUITE)
expectation_suite = context.suites.add(expectation_suite)

# -- 2.1. Define table level expectations
columns = list(df_configure.columns)
expectation = gx.expectations.ExpectTableColumnsToMatchSet(column_set=columns)
expectation_suite.add_expectation(expectation)

# -- 2.2. Define column level expectations
expectation = gx.expectations.ExpectColumnValuesToNotBeNull(column="VendorID")
expectation_suite.add_expectation(expectation)

expected_values = [1, 2]
expectation = gx.expectations.ExpectColumnValuesToBeInSet(
    column="VendorID", value_set=expected_values
)
expectation_suite.add_expectation(expectation)

# -- 2.3. Evaluate results on test dataset
batch_parameters = {"dataframe": df_configure}
batch = batch_definition.get_batch(batch_parameters=batch_parameters)
validation_results = batch.validate(expectation_suite)

# -- 3. Bundle suite and batch into validation definition and checkpoint w/ bundled
# --    actions for easy execution later
validation_definition = gx.ValidationDefinition(
    data=batch_definition, suite=expectation_suite, name=NAME_VALIDATION_DEF
)
validation_definition = context.validation_definitions.add(validation_definition)

action_list = [
    gx.checkpoint.UpdateDataDocsAction(
        name="update_all_data_docs",
    ),
]
checkpoint = gx.Checkpoint(
    name=NAME_CHECKPOINT,
    validation_definitions=[validation_definition],
    actions=action_list,
    result_format={
        "result_format": "COMPLETE",
    },
)
context.checkpoints.add(checkpoint)

# -- 4. Run checkpoint to validate if everything works properly
runid = gx.RunIdentifier(run_name="Configuration run")
results = checkpoint.run(batch_parameters=batch_parameters, run_id=runid)
```

**Expected behavior**
I would expected that asset name of the data asset used to invoke the checkpoint over to show up in the asset name section, but instead it is empty.

**Environment (please complete the following information):**
 - Operating System: MacOS
 - Great Expectations Version: 1.0.1
 - Data Source: Pandas
 - Cloud environment: N/A
 
[yellow_tripdata_2021-11.csv](https://github.com/user-attachments/files/16885718/yellow_tripdata_2021-11.csv)

**Comment by Kilo59** [comment]
Apologies this was fixed in [0.18.14](https://github.com/great-expectations/great_expectations/releases/tag/0.18.14) after we forked our development branches for 1.0.

- Original issue https://github.com/great-expectations/great_expectations/issues/8790
- Original fix https://github.com/great-expectations/great_expectations/pull/9950

This fix for 1.0 is being worked on here
- https://github.com/great-expectations/great_expectations/pull/9953

**Comment by Kilo59** [comment]
@jschra This should be fixed in `1.0.2`

https://github.com/great-expectations/great_expectations/releases/tag/1.0.2
